### PR TITLE
Cleanup rules.mk for 32U4 keyboards, R-S

### PR DESCRIPTION
**Pull request by fauxpark**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Duckle29** [comment]
Why is bootmagic disabled for `keyboards/suihankey/split` ?  :)

**Comment by fauxpark** [comment]
It's not, it should be grabbing the build options from the top-level rules.mk.

File: keyboards/suihankey/split/alpha/rules.mk [hunk 1]
```diff
@@ -1,82 +1 @@
...
-#   Atmel DFU loader    4096
-#   LUFA bootloader     4096
-#   USBaspLoader        2048
-# OPT_DEFS += -DBOOTLOADER_SIZE=4096
-
-
-# Build Options
-#   change yes to no to disable
-#
-BOOTMAGIC_ENABLE = lite    # Virtual DIP switch configuration(+1000)
```
Comment (Duckle29) [inline keyboards/suihankey/split/alpha/rules.mk]: Why is bootmagic disabled for this keyboard?

Comment (Duckle29) [reply keyboards/suihankey/split/alpha/rules.mk]: I moved it to this review thing, still new to it :) And I see, there's a further level of rules.mk :)
Thanks

**Pull request merged**